tests/e2e/test_api.py

```diff
@@ -444,6 +443,17 @@ def test_query_filter() -> None:
         assert "foo" not in response_text
         assert "bar" not in response_text
 
+        data_collection_container_name = "ols-sidecar-user-data-collector"
+        pod_name = cluster_utils.get_single_existing_pod_name()
+
+        container_log = cluster_utils.get_container_log(
+            pod_name, data_collection_container_name
+        )
+
+        # Check if redacted words are not logged in the container log
+        assert "foo" not in container_log
+        assert "bar" not in container_log
```

Comment:
please do a "positive" test as well, meaning you should make sure the expected(substituted) content (in this case text like `deploy` and `openshift`) is in the log.  Otherwise this test can pass just based on us (for some reason) not getting any log data back.

Reply:
looks like you're going to need a more exact match here because the test is currently failing by matching other log entries that contain "foo":

```
{assert 'foo' not in '2024-06-18 ...ed":false}\n'
  
  'foo' is contained here:
    ter(name='foo_filter', pattern='\\b(?:foo)\\b', replace_with='deployment'), QueryFilter(name='bar_filter', pattern='\\b(?:bar)\\b', replace_with='openshift')]    
```